### PR TITLE
Add Support For AWS Dex Basic-Authentication

### PR DESCRIPTION
**Pull request by PatrickXYS**
This is for internal review.

### CODE REVIEW

File: pkg/kfapp/aws/aws.go [hunk 1]
```diff
@@ -776,6 +816,15 @@ func (aws *Aws) Apply(resources kftypes.ResourceEnum) error {
 		}
 	}
 
+	// 6. Setup Dex basic-authentication
+	if err := aws.setupDex(); err != nil {
```
Comment (Jeffwan) [inline pkg/kfapp/aws/aws.go:820]: I think we only do this for basic setup. Let's make sure cognito path will skip this

File: pkg/kfapp/aws/aws.go [hunk 2]
```diff
@@ -1107,3 +1174,174 @@ func (aws *Aws) setupOIDC() error {
...
+	}
+
+	// check if we need it or not
+	// Install dex Istio from dex Istio Manifests
+	if err := applyManifests(aws.dexIstioManifests); err != nil {
+		return internalError(errors.WithStack(err))
+	}
+
+	// Get Kubeflow and Dex Endpoints
+	kfEndpoint, oidcEndpoint, err := getEndpoints(aws.k8sClient)
```
Comment (Jeffwan) [inline pkg/kfapp/aws/aws.go:1213]: BTW, what's the endpoint here? It's ALB or ELB? existing.go will create LoadBalancer k8s service which is ELB by default. Can we check if we are able to use ALB?

File: pkg/kfapp/aws/aws.go [hunk 3]
```diff
@@ -1107,3 +1174,174 @@ func (aws *Aws) setupOIDC() error {
...
+		return internalError(errors.WithStack(err))
+	}
+	if err := createSelfSignedCerts(aws.k8sClient, kfEndpointURL.Hostname()); err != nil {
+		return internalError(errors.WithStack(err))
+	}
+
+	// Get the kubeflow user to add
+	// TODO(yanniszark): get this from a plugin struct eventually (https://github.com/kubeflow/kubeflow/issues/3529)
+	log.Info("Getting the Kubeflow User")
+	kubeflowUser, err := getKubeflowUser()
```
Comment (Jeffwan) [inline pkg/kfapp/aws/aws.go:1230]: This user should come from our basic auth in AWSPlugin

File: pkg/kfapp/aws/aws.go (hunk 1 above)
Comment (PatrickXYS) [reply pkg/kfapp/aws/aws.go:820]: Sure, I added logic to skip this when using cognito.

File: pkg/kfapp/aws/aws.go (hunk 3 above)
Comment (PatrickXYS) [reply pkg/kfapp/aws/aws.go:1230]: Yeah I added logic in basicauth to add username and password.

**Pull request closed without merging**